packages/provider/wrapper.js

```diff
@@ -32,6 +32,43 @@ module.exports = {
       const originalSendAsync = provider.sendAsync.bind(provider);
       provider.sendAsync = this.send(originalSendAsync, preHook, postHook);
     }
+
+    // EIP-1193 provider wrapping
+    if (
+      typeof provider.request === "undefined" &&
+      typeof provider.send !== "undefined"
+    ) {
+      provider.request = async function ({ method, params }) {
```

Comment:
Is there any way to clean this wrapper up while you're at it? Not necessary, but I think we'd be a lot happier in the future if this whole function was divided into two equal parts (e.g., it could defer to separate wrapper functions for both legacy and EIP-1193 cases)

Reply:
It is a nice idea to clean up the wrapper function. I will look into it. Thanks for the suggestion @gnidan.